java.lang.IllegalStateException: Need to call #setCharacterLists first

I have  initialised CharacterList in onCreate() method . It is still giving the following error:

Caused by: java.lang.IllegalStateException: Need to call #setCharacterLists first.
        at com.robinhood.ticker.TickerColumnManager.setText(TickerColumnManager.java:71)
        at com.robinhood.ticker.TickerView.setText(TickerView.java:299)
        at com.robinhood.ticker.TickerView.init(TickerView.java:180)

--- a/ticker/src/main/java/com/robinhood/ticker/TickerView.java
+++ b/ticker/src/main/java/com/robinhood/ticker/TickerView.java
@@ -266,7 +266,7 @@ public class TickerView extends View {
     public void setCharacterLists(String... characterLists) {
         columnManager.setCharacterLists(characterLists);
         if (pendingTextToSet != null) {
-            setText(pendingTextToSet);
+            setText(pendingTextToSet, false);
             pendingTextToSet = null;
         }
     }

--- a/ticker/src/main/java/com/robinhood/ticker/TickerView.java
+++ b/ticker/src/main/java/com/robinhood/ticker/TickerView.java
@@ -86,6 +86,8 @@ public class TickerView extends View {
     private long animationDurationInMillis;
     private Interpolator animationInterpolator;
     private boolean animateMeasurementChange;
+    // pending text set from XML because we didn't have a character list initially
+    private String pendingTextToSet;
 
     public TickerView(Context context) {
         super(context);
@@ -177,7 +179,11 @@ public class TickerView extends View {
                 }
         }
 
-        setText(styledAttributes.text, false);
+        if (isCharacterListsSet()) {
+            setText(styledAttributes.text, false);
+        } else {
+            this.pendingTextToSet = styledAttributes.text;
+        }
 
         arr.recycle();
 
@@ -259,6 +265,10 @@ public class TickerView extends View {
      */
     public void setCharacterLists(String... characterLists) {
         columnManager.setCharacterLists(characterLists);
+        if (pendingTextToSet != null) {
+            setText(pendingTextToSet);
+            pendingTextToSet = null;
+        }
     }
 
     /**

